Modules/Loadable/VolumeRendering/MRML/vtkMRMLVolumeRenderingDisplayNode.cxx

```diff
@@ -333,3 +337,101 @@ void vtkMRMLVolumeRenderingDisplayNode::ProcessMRMLEvents(vtkObject *caller,
     this->InvokeEvent(event);
   }
 }
+
+//----------------------------------------------------------------------------
+bool vtkMRMLVolumeRenderingDisplayNode::IsClippingShortcutValid(vtkMRMLMessageCollection* userMessages/*=nullptr*/)
+{
+  bool shortcutValid = true;
+  if (this->GetCroppingEnabled())
+  {
+    if (userMessages)
+    {
+      userMessages->AddMessage(vtkCommand::MessageEvent, vtkMRMLTr("vtkMRMLVolumeRenderingDisplayNode", "Fast clipping cannot be used with cropping enabled."));
+    }
+    shortcutValid = false;
+  }
+
+  if (this->GetClippingSoftEdgeVoxels() > 0.0)
+  {
+    if (userMessages)
+    {
+      userMessages->AddMessage(vtkCommand::MessageEvent, vtkMRMLTr("vtkMRMLVolumeRenderingDisplayNode", "Fast clipping cannot be used with soft edge voxels."));
+    }
+    shortcutValid = false;
+  }
+
+  if (!this->IsClippingShortcutValid(this->GetClipNode(), userMessages))
+  {
+    shortcutValid = false;
+  }
+
+  return shortcutValid;
+}
+
+//---------------------------------------------------------------------------
+bool vtkMRMLVolumeRenderingDisplayNode::IsClippingShortcutValid(vtkMRMLClipNode* clipNode, vtkMRMLMessageCollection* userMessages/*=nullptr*/)
```

Comment:
"Valid shortcut" name could be somewhat confusing. Something like "fast clipping available" would be more clear.

```suggestion
bool vtkMRMLVolumeRenderingDisplayNode::IsFastClippingAvailable(vtkMRMLClipNode* clipNode, vtkMRMLMessageCollection* userMessages/*=nullptr*/)
```